### PR TITLE
Delete deprecated cli flags

### PR DESCRIPTION
**Pull request by chong-he**
Delete deprecated cli flags as summarised below:

|No. | Deleted flag | CLI | Remark |
| --- | --- | --- | --- |
|1| --spec| lighthouse bn --help||
|2| --eth1-endpoint | lighthouse bn --help |  |
|3| --eth1-endpoints | lighthouse bn --help |  |
|4| --beacon-node| lighthouse vc --help||
|5| --server | lighthouse vc --help ||
|6|--delete-lockfiles| lighthouse vc --help||
|7| --allow-unsynced| lighthouse vc --help ||
|8| --strict-fee-recipient| lighthouse vc --help||
|9| --merge | lighthouse bn --help | hidden flag|
|10| --count-unrealized| lighthouse bn --help| hidden flag|
|11| --count-unrealized-full| lighthouse bn --help|hidden flag|
|12| --http-disable-legacy-spec| lighthouse bn --help |hidden flag|
|13| --minify| lighthouse account validator slashing-protection import/export --help ||

### CODE REVIEW

File: account_manager/src/validator/slashing_protection.rs [hunk 1]
```diff
@@ -102,23 +79,10 @@ pub fn cli_run<T: EthSpec>(
...
                 .map_err(|e| format!("Error parsing file for import: {:?}", e))?;
             eprintln!(" [done].");
 
-            if let Some(minify) = minify {
-                eprintln!(
-                    "WARNING: --minify flag is deprecated and will be removed in a future release"
-                );
-                if minify {
-                    eprint!("Minifying input file for faster loading");
-                    interchange = interchange
```
Comment (michaelsproul) [inline account_manager/src/validator/slashing_protection.rs:115]: by deleting this line we are no longer mutating `interchange`, so it doesn't need to be mutable

File: lcli/src/main.rs [hunk 2]
```diff
@@ -276,14 +276,6 @@ fn main() {
...
             SubCommand::with_name("eth1-genesis")
                 .about("Listens to the eth1 chain and finds the genesis beacon state")
-                .arg(
-                    Arg::with_name("eth1-endpoint")
-                        .short("e")
-                        .long("eth1-endpoint")
-                        .value_name("HTTP_SERVER")
-                        .takes_value(true)
-                        .help("Deprecated. Use --eth1-endpoints."),
-                )
```
Comment (michaelsproul) [inline lcli/src/main.rs]: Lets keep this, because it seems since the merge we only support 1 eth1-endpoint

I think we should actually deprecate eth1-endpoints and _undeprecate_ eth1-endpoint, but we can do that in another PR

Comment (chong-he) [reply lcli/src/main.rs]: Add back in [0d17bd0](https://github.com/sigp/lighthouse/pull/4906/commits/0d17bd0ec28370459e32b33e5631ac0a847fd60f)

File: account_manager/src/validator/slashing_protection.rs (hunk 1 above)
Comment (chong-he) [reply account_manager/src/validator/slashing_protection.rs:115]: Thank you

**Review by michaelsproul — APPROVED** [review]
This is awesome, thanks CK!

Feels good to get rid of all this old baggage!
